server/src/main/java/io/crate/execution/engine/export/FileOutput.java

```diff
@@ -35,5 +36,8 @@
      *
      * @throws IOException in case the Output can't be created (e.g. due to file permission errors or something like that)
      */
-    OutputStream acquireOutputStream(Executor executor, URI uri, WriterProjection.CompressionType compressionType) throws IOException;
+    OutputStream acquireOutputStream(Executor executor,
+                                     URI uri,
+                                     WriterProjection.CompressionType compressionType,
+                                     @Nullable String protocolSetting) throws IOException;
```

Comment:
What do you think of using a settings map or some else that is more generic here? `protocol` is exclusive to S3 and the `FileOutput` interface as generic.

Otherwise we may end up having a superset of options here that are only supported by subsets of implementations.
It also shouldn't be necessary to touch the interface each time we'd want to add support for a new scheme or have a plugin with custom options.

Reply:
yes, I agree will work on it.

Reply:
We could also have some kind of `parseWithOptions` functions on the `CopyPlugin`, so that each plugin can have custom logic to parse & validate additional `WITH` options that aren't supported by the framework

Reply:
Please check 2240304 for the rough idea. If this seems okay, the next step is to include all other withClause options like `compressionType`. Also, it looks like this map could be created all the way back from `CopyToPlan`.